UOP1_Project/Assets/Scripts/DesignPatterns/StateMachine/StateMachine.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class StateMachine
```

Comment:
It would be nice if this could be visualized in the Inspector somehow. Seeing what state a statemachine is currently in is very often a useful thing to be able to see.

Reply:
Yeah I agree with you on that, since right now it's not really possible to be visualized at all.

Reply:
if you make a public getter for currentState you can then attach some visualizer easily enough